core/sr-primitives/src/generic/unchecked_mortal_extrinsic.rs

```diff
@@ -76,19 +77,20 @@ where
 	AccountId: Member + MaybeDisplay,
 	BlockNumber: SimpleArithmetic,
 	Hash: Encode,
-	Context: Lookup<Source=Address, Target=AccountId>
+	Context: Lookup<Source=Address, Target=AccountId, Error=&'static str>
 		+ CurrentHeight<BlockNumber=BlockNumber>
 		+ BlockNumberToHash<BlockNumber=BlockNumber, Hash=Hash>,
 {
 	type Checked = CheckedExtrinsic<AccountId, Index, Call>;
+	type Error = Error;
 
-	fn check(self, context: &Context) -> Result<Self::Checked, &'static str> {
+	fn check(self, context: &Context) -> Result<Self::Checked, Error> {
 		Ok(match self.signature {
 			Some((signed, signature, index, era)) => {
 				let current_u64 = context.current_height().saturated_into::<u64>();
 				let h = context.block_number_to_hash(era.birth(current_u64).saturated_into())
-					.ok_or("transaction birth block ancient")?;
-				let signed = context.lookup(signed)?;
+					.ok_or(Error::Unknown("transaction birth block ancient"))?;
```

Comment:
As you have an `Into<Error>`, you should not require `Error::Unknown` here.

Reply:
Remove `Error::Unknown` doesn't compile